saleor/order/utils.py

```diff
@@ -275,15 +275,25 @@ def change_order_line_quantity(user, line, old_quantity, new_quantity):
 
     quantity_diff = old_quantity - new_quantity
 
-    # Create the removal event
+    # Create the added / removed products event based on order status
     if quantity_diff > 0:
-        events.draft_order_removed_products_event(
-            order=line.order, user=user, order_lines=[(quantity_diff, line)]
-        )
+        if line.order.status == OrderStatus.DRAFT:
+            events.draft_order_removed_products_event(
+                order=line.order, user=user, order_lines=[(quantity_diff, line)]
+            )
+        else:
+            events.unconfirmed_order_removed_products_event(
+                order=line.order, user=user, order_lines=[(quantity_diff, line)]
+            )
```

Comment:
Refactored. Now we use a signle event type on `OrderEvent`, instead of `draft_added_products` and `draft_removed_products`, we now have `added_products` and `removed_products`. Please have a look on `saleor/saleor/order/migrations/0097_auto_20210125_0749.py` - this is a combined migration which also changes existing DB events.